homeassistant/components/knx/__init__.py

```diff
@@ -445,27 +453,28 @@ async def service_exposure_register_modify(self, call):
 
         if group_address in self.service_exposures:
             replaced_exposure = self.service_exposures.pop(group_address)
+            assert replaced_exposure.device is not None
             _LOGGER.warning(
                 "Service exposure_register replacing already registered exposure for '%s' - %s",
                 group_address,
                 replaced_exposure.device.name,
             )
             replaced_exposure.shutdown()
-        exposure = create_knx_exposure(self.hass, self.xknx, call.data)
+        exposure = create_knx_exposure(self.hass, self.xknx, call.data)  # type: ignore[arg-type]
```

Comment:
`call` is a `homeassistant.helpers.typing.ServiceCallType`. 
Why is `call.data` not a `homeassistant.helpers.typing.ServiceDataType`, but a `MappingProxyType[Any, Any]` here?
`create_knx_exposure` is expecting a `ConfigType` (`Dict[str, Any]`) as 3rd argument. Should I change this for  a `Union[ConfigType,  MappingProxyType[Any, Any]]` ?

Reply:
Use the `ServiceCall` from core intead.

Reply:
Result is the same.
https://github.com/home-assistant/core/blob/88b5eff726155d44064e4c954e00ef8e326c2046/homeassistant/helpers/typing.py#L13